trin-utils/build.rs

```diff
@@ -1,11 +1,29 @@
+use std::env;
 use std::process::Command;
+
 fn main() {
     let output = Command::new("git")
         .args(["rev-parse", "HEAD"])
         .output()
         .expect("Unable to get git hash");
-    let git_hash = String::from_utf8(output.stdout)
-        .expect("git rev-parse output must be a valid utf8 encoding");
+    let output = String::from_utf8(output.stdout)
+        .expect("git rev-parse output must be a valid utf8 encoding")
+        .replace('\n', "");
+    let git_hash = match output.len() {
+        40 => output,
+        _ => {
+            // If the git hash is not 40 characters, then we are probably in a
+            // docker build context, in which case we access the git hash via the
+            // environment variable set in the Docker build command
+            match env::var("GIT_HASH") {
+                Ok(val) => match val.len() {
+                    40 => val,
+                    _ => "".to_string(),
```

Comment:
IIUC we should never encounter this `match` arm, so it could be useful to log some warning here.

Reply:
There's a case when `docker build ...` is used *without* passing in a `--build-arg GIT_HASH=$CIRCLE_SHA1` arg. Then, the default `GIT_HASH=unknown` env var will be set in the Dockerfile. Which will lead to this `match` arm, and in which case we want to set the rustc env var to `""` to have some sort of consistent behavior for our `get_trin_version()` util

Also, according to the [docs](https://doc.rust-lang.org/cargo/reference/build-scripts.html#outputs-of-the-build-script) - `The output of the script is hidden from the terminal during normal compilation.` unless you use `"very verbose" with the -vv flag.`. Which I'd expect to rarely happen if ever, so I'm not sure how useful a logged warning is in this case